lib/init.luau

```diff
@@ -94,17 +94,18 @@ function Net:__iter(filter: (k: number, v: any) -> (number?, any?)?, returnFilte
 	local snapshot = self._bridge:snapshot()
 	local i = 0
 
-	-- Filters take in a key-value pair, the filter can remove this pair by returning ``k, nil``.
-	if filter then
-		for k, v in snapshot do
-			v = filter(v)
-
-			if v then
-				snapshot[k] = v
-			else
-				table.remove(snapshot, k)
+	-- Filters take in a key-value pair, the filter can remove this pair by not adding it to the filtered_snapshot array.
+	do
```

Comment:
I don't think we need this scope here, let's avoid the unnecessary nesting. If this is here for a reason let me know.